api/role.go

```diff
@@ -47,3 +48,66 @@ func (api *API) GetRoleHandler(w http.ResponseWriter, req *http.Request) {
 	log.Event(ctx, "getRole Handler: Successfully retrieved role", log.INFO, logdata)
 
 }
+
+//GetRolesHandler is a handler that gets all roles from MongoDB
+func (api *API) GetRolesHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+
+	//get limit from query parameters, or default value
+	limit, err := getPaginationQueryParameter(req.URL.Query(), "limit", api.defaultLimit)
+	if err != nil {
+		log.Event(ctx, "failed to obtain limit from request query parameters", log.ERROR)
```

Comment:
Consider including the error in the log output as this will contain specific details about why it failed - helpful to anyone looking into this if its not working.